contracts/cw1-subkeys/src/contract.rs

```diff
@@ -505,714 +651,1060 @@ mod tests {
-        execute(deps.as_mut(), mock_env(), info, setup_perm_msg3).unwrap();
+            let rsp = execute(
+                deps.as_mut(),
+                env,
+                owner,
+                ExecuteMsg::IncreaseAllowance {
+                    spender: SPENDER1.to_owned(),
+                    amount: coin(3, TOKEN1),
+                    expires: Some(Expiration::AtHeight(10)),
+                },
+            )
+            .unwrap();
+
+            assert_eq!(rsp.messages, vec![]);
+            assert_eq!(rsp.events, vec![]);
+            assert_eq!(rsp.data, None);
+
+            assert_eq!(
+                query_all_allowances(deps.as_ref(), None, None)
+                    .unwrap()
+                    .canonical(),
+                AllAllowancesResponse {
+                    allowances: vec![AllowanceInfo {
+                        spender: SPENDER1.to_owned(),
+                        balance: NativeBalance(vec![coin(4, TOKEN1)]),
+                        expires: Expiration::AtHeight(10),
+                    }]
+                }
+                .canonical()
+            );
+        }
 
-        // let's try pagination
-        let permissions = query_all_permissions(deps.as_ref(), None, Some(2))
-            .unwrap()
-            .permissions;
-        assert_eq!(2, permissions.len());
-        assert_eq!(
-            permissions[0],
-            PermissionsInfo {
-                spender: spender1.into(),
-                permissions: god_mode,
-            }
-        );
-        assert_eq!(
-            permissions[1],
-            PermissionsInfo {
-                spender: spender2.to_string(),
-                permissions: noob_mode,
-            }
-        );
+        #[test]
+        fn new_token_on_existing_spender() {
+            let Suite {
+                mut deps, owner, ..
+            } = SuiteConfig::new()
+                .with_allowance(SPENDER1, coin(1, TOKEN1))
+                .init();
 
-        // now continue from after the last one
-        let permissions = query_all_permissions(deps.as_ref(), Some(spender2.into()), Some(2))
-            .unwrap()
-            .permissions;
-        assert_eq!(1, permissions.len());
-        assert_eq!(
-            permissions[0],
-            PermissionsInfo {
-                spender: spender3.into(),
-                permissions: noob_mode,
-            }
-        );
-    }
+            let rsp = execute(
+                deps.as_mut(),
+                mock_env(),
+                owner,
+                ExecuteMsg::IncreaseAllowance {
+                    spender: SPENDER1.to_owned(),
+                    amount: coin(3, TOKEN2),
+                    expires: None,
+                },
+            )
+            .unwrap();
+
+            assert_eq!(rsp.messages, vec![]);
+            assert_eq!(rsp.events, vec![]);
+            assert_eq!(rsp.data, None);
+
+            assert_eq!(
+                query_all_allowances(deps.as_ref(), None, None)
```

Comment:
since there is only one SPENDER here, why not use `query_allowance`?
It seems like it would be a bit simpler.

Reply:
Because as I am touching allowance, I want to be sure that only my specific allowance changed, and no side effect occurred. It is surprisingly almost the same (just one encapsulation more), and more generic test.